lib/PlatoonService/src/PlatoonUtils.cpp

```diff
@@ -117,13 +117,27 @@ bool PlatoonUtils::areWaypointsEqual(const Waypoint& waypoint1, const Waypoint&
 
 int32_t PlatoonUtils::calculateEquivalentHeading(int32_t targetHeading, int32_t referenceHeading)
 {
-    int32_t equivalentHeading = targetHeading;
+    int32_t equivalentHeading = targetHeading % FP_2PI();
+
+    if (FP_2PI() == referenceHeading)

```

Comment:
2PI and -2PI are the same as 0, so here I think the referenceHeading should be set to 0, because the equivalentHeading is already 0 by modulo.

Reply:
Agreed, but this covers the case in which the measured heading has not performed a modulo operation. If this is the case, reseting the angle to 0 would cause it to turn the wrong way. The reference heading cannot be modified/set to 0 with modulo, therefore this case must be handled.

Reply:
I don't think you find it in a different range, because 0..2PI is the whole range, as well as -PI...PI contains it. But we should use internally always the same range to not confuse ourself. :-)

Reply:
One example is when reference is 2PI and target it PI/2. The closest equivalent angle is reached counter-clockwise. However as the angle PI/2 < 2PI, the robot will turn clockwise all the way to PI/2 instead. This is what the different ranges here are trying to prevent

Reply:
I am talking about the ranges of the input values only. :-)
If you change the range inside the function, please comment why and to which range. This makes it more understandable.